pkg/sbom/cyclonedx/unmarshal.go

```diff
@@ -405,3 +407,11 @@ func toTrivyCdxComponent(component cdx.Component) ftypes.Component {
 		PackageURL: component.PackageURL,
 	}
 }
+
+func getPackageName(typ string, component cdx.Component) string {
+	// Jar uses `Group` field for `GroupID`
+	if typ == packageurl.TypeMaven && component.Group != "" {
+		return fmt.Sprintf("%s:%s", component.Group, component.Name)
+	}
+	return component.Name
+}
```

Comment:
I think it should be done here. What do you think?
https://github.com/aquasecurity/trivy/blob/df61fd468b7b11bd9c409fede1304f4c1a7d9dcf/pkg/purl/purl.go#L73-L85

Reply:
Or is PURL missing groupID?

Reply:
we use component.Name after this PR - https://github.com/aquasecurity/trivy/pull/4306/files#diff-caf585364b1a95f5cb5dbda210a5ca931fad058915ba586540c8aa32a7883311R346

But maybe it's better to use the purl name for the jars. Wdyt?

Reply:
Oh, I just remember the change. OK, so we can go with your change.